pkg/kubernetes/deploy.js

```diff
@@ -73,6 +80,220 @@ define([
+                    .fail(function(ex, response) {
+                        /* Display the error appropriately in the dialog */
+                        $("#deploy-app-dialog").dialog("failure", ex);
+                    });
+
+                    /* Display a spinner while tmp folder is happening */
+                    $("#deploy-app-dialog").dialog("wait", promise1);
+
+            });
+
+        /* Display a spinner while vaidation is happening */
+        $("#deploy-app-dialog").dialog("wait", promise);
+    }
+
+
+    function validate_app_fields() {
+        var dfd = $.Deferred();
+        var ex, fails = [];
+        var label_input = get_fields(convertToKV(answerfile));
+        var fields = { };
+        var tmp = nulecule_client.get_answers();
+
+        for(var x in label_input) {
+            var li = label_input[x];
+            var input_id = "#" + li.input_val;
+            var label_id = li.label_val;
+            var app_data = get_app_from_label(label_id);
+            var input_value = $(input_id).val();
+
+            tmp[app_data.app_name][app_data.key] = input_value;
+
+            var label_value = $('#deploy-app-dialog').find('label[for="' + label_id + '"]').text().trim();
+            if (!input_value)
+                ex = new Error(label_value + " cannot be empty.");
+
+            if (ex) {
+                ex.target = input_id;
+                fails.push(ex);
+                ex = null;
+            } 
+        }
+
+        var ns = $("#deploy-app-namespace").val();
+        if (!ns)
+            ex = new Error(_("Namespace cannot be empty."));
+        else if (!/^[a-z0-9]+$/i.test(ns))
+            ex = new Error(_("Please provide a valid namespace."));
+        if (ex) {
+            ex.target = "#deploy-app-namespace-group";
+            fails.push(ex);
+            ex = null;
+        } else {
+            tmp.general.namespace = ns;
+        }
+
+        if (fails.length) {
+            dfd.reject(fails);
+        } else { 
+            dfd.resolve();
+        }
+
+        nulecule_client.set_answers(tmp);
+        return dfd.promise();
+    }
+
+
+    function get_fields(applist) {
+        var labels = [];
+        for(var x in applist) {
+            var app = applist[x];
+            var prefix = app.app_name;
+            var lsuffix = "label";
+            var isuffix = "text";
+
+            for(var y in app.app_params) {
+                var param = app.app_params[y];
+                var label = prefix + "-" + param.param_key + "-" + lsuffix;
+                var input = prefix + "-" + param.param_key + "-" + isuffix;
+                var tmp = { "label_val" : label, "input_val" : input };
+                labels.push(tmp);
+            }
+        }
+        return labels;
+    }
+
+
+    /*
+     * Get app_name and KEY from label
+     */
+    function get_app_from_label(label) {
+        var app_name = "";
+        var key = "";
+        key = label.split("-label")[0].split("-");
```

Comment:
Why do you set a variable to an empty string right before setting it to another value?